app/src/main/java/com/google/wear/whereami/WhereAmIActivity.kt

```diff
@@ -40,30 +47,48 @@ class WhereAmIActivity : FragmentActivity() {
         setContentView(R.layout.where_am_i_activity)
         textView = findViewById(R.id.text)
 
-        locationViewModel = LocationViewModel(applicationContext)
+        locationViewModel = (this.applicationContext as WhereAmIApplication).locationViewModel
 
-        lifecycleScope.launch {
+        lifecycleScope.launchWhenCreated {
             checkPermissions()
 
-            val location = locationViewModel.readLocationResult()
+            val flow = locationViewModel.store.stream(StoreRequest.cached(Current, refresh = false))
 
-            if (location is ResolvedLocation) {
-                textView.text = getString(
-                    R.string.address_as_of_time_activity,
-                    getAddressDescription(location),
-                    getTimeAgo(location.location.time)
-                )
-            } else {
-                textView.setText(R.string.location_error)
+            flow.collect {
```

Comment:
I vaguely remember there's something like flow.filterNulls or something that could make this simpler... or maybe that was RxJava?